uitools/import/Esri/ArcGISRuntime/Toolkit/Callout.qml

```diff
@@ -404,692 +333,349 @@ Item {
-                        left: parent.left
-                        top: parent.top
+                startY: {
+                    switch(internal.leaderPosition) {
+                    case Callout.LeaderPosition.UpperLeft:
+                    case Callout.LeaderPosition.Top:
+                    case Callout.LeaderPosition.UpperRight:
+                        return tail.startY + background.border.width / 2;
+                    case Callout.LeaderPosition.LowerRight:
+                    case Callout.LeaderPosition.Bottom:
+                    case Callout.LeaderPosition.LowerLeft:
+                        return tail.startY - background.border.width / 2;
+                    case Callout.LeaderPosition.Left:
+                    case Callout.LeaderPosition.Right:
+                        return tail.startY + leaderXOffset;
                     }
-                    height: calloutHeight
-                    width: calloutWidth
-                    columns: 3
-                    rows: 2
-                    columnSpacing: 7
-
-                    Rectangle {
-                        id: imageRect
-                        width: 40
-                        height: width
-                        color: "transparent"
-                        Layout.rowSpan: 2
-                        Layout.alignment: Qt.AlignHCenter | Qt.AlignVCenter
-
-                        Image {
-                            id: image
-                            source: calloutData ? calloutData.imageUrl : ""
-                            width: parent.width
-                            height: width
-                            fillMode : Image.PreserveAspectFit
+                }
+                PathLine {
+                    relativeX: {
+                        switch(internal.leaderPosition) {
+                        case Callout.LeaderPosition.UpperLeft:
+                        case Callout.LeaderPosition.Top:
+                        case Callout.LeaderPosition.UpperRight:
+                        case Callout.LeaderPosition.LowerRight:
+                        case Callout.LeaderPosition.Bottom:
+                        case Callout.LeaderPosition.LowerLeft:
+                            return root.leaderWidth - hideLine.leaderXOffset * 2;
+                        case Callout.LeaderPosition.Left:
+                        case Callout.LeaderPosition.Right:
+                            return 0;
                         }
                     }
-
-                    Label {
-                        id: title
-                        text: calloutData ? calloutData.title : ""
-                        wrapMode: Text.NoWrap
-                        renderType: Text.NativeRendering
-                        color: titleTextColor
-                        font {
-                            pixelSize: 11
-                            family: "sanserif"
+                    relativeY: {
+                        switch(internal.leaderPosition) {
```

Comment:
❓ None of the switches have default cases. Should they?

Reply:
Good point, I've jiggled things around so `default` always matches the `LeaderPosition.Bottom` case.